zerver/tests/test_signup.py

```diff
@@ -2046,3 +2046,87 @@ def test_login_under_subdomains(self):
+        prereg_user_one.save() 
+        expired_datetime = timezone_now() - datetime.timedelta(days=(days_to_activate+1))
+        prereg_user_two = PreregistrationUser(email="TestTwo@zulip.com", referred_by=user_profile)
+        prereg_user_two.save()
+        PreregistrationUser.objects.filter(id=prereg_user_two.id).update(invited_at=expired_datetime)
+        prereg_user_three = PreregistrationUser(email="TestThree@zulip.com", referred_by=user_profile, status=active_value)
+        prereg_user_three.save()
+        
+        result = self.client_get("/json/invites")
+        self.assertEqual(result.status_code, 200)
+        self.assert_in_success_response("TestOne@zulip.com", result)
+        self.assert_not_in_success_response(["TestTwo@zulip.com", "TestThree@zulip.com"], result)
+        
+    def test_successful_delete_invitation(self):
+        """
+        A DELETE call to /json/invites/<ID> should delete the invite and any scheduled invitation reminder email
+        """
+        
+        self.login(self.example_email("iago"))
+        user_profile = self.example_user("iago")
+        
+        invitee = "DeleteMe@zulip.com"
+
+        prereg_user = PreregistrationUser(email=invitee, referred_by=user_profile)
+        prereg_user.save()
+        
+        delete_id = prereg_user.id
+        
+        result = self.client_delete('/json/invites/'+str(delete_id))
+        self.assertEqual(result.status_code, 200)
+        error_result = self.client_delete('/json/invites/'+str(delete_id))
+        self.assert_json_error(error_result, "Cannot revoke the invitation, the invite was not found.")
+        
+        self.assertRaises(ScheduledEmail.DoesNotExist,
+                              lambda: ScheduledEmail.objects.get(address__iexact=invitee,
+                                          type=ScheduledEmail.INVITATION_REMINDER))
+    
+    def test_successful_resend_invitation(self):
+        """
+        A POST call to /json/invites/resend/<ID> should send an invitation reminder email
+        and delete any scheduled invitation reminder email
+        """
+        
+        self.login(self.example_email("iago"))
+        user_profile = self.example_user("iago")
+        invitee = "ResendMe@zulip.com"
+        
+        prereg_user = PreregistrationUser(email=invitee, referred_by=user_profile)
+        prereg_user.save()
+        resend_id = prereg_user.id
+        
+        result = self.client_post('/json/invites/resend/'+str(resend_id))
+
+        self.assertEqual(result.status_code, 200)
+        faulty_result = self.client_post('/json/invites/resend/'+str(9999), {"prereg_id": 9999})
+        self.assert_json_error(faulty_result, "Cannot resend the invitation email, the invite was not found.")
+        
+        mail_test = InviteUserTest()
+        self.assertTrue(find_key_by_email(invitee))
+        mail_test.check_sent_emails([invitee], custom_from_name="Zulip")
```

Comment:
Rather than doing this hackery with `InviteUserTest`, you should just extract a `InviteUserBase` class that just has the couple functions you need, and make both this and `InviteUserTest`  subclass it to get access to this function.

Do this in a preparatory commit at the start of your branch, since we can merge that early.